libraries/joomla/form/rules/equals.php

```
@@ -44,13 +44,13 @@ public function test($element, $value, $group = null, $input = null, $form = nul
 		// Check that a validation field is set.
 		if (!$field)
 		{
-			return new JException(JText::sprintf('JLIB_FORM_INVALID_FORM_RULE', get_class($this)));
+			return new JException(sprintf('Invalid Form Rule: :%s', get_class($this)));
 		}
 
 		// Check that a valid JForm object is given for retrieving the validation field value.
 		if (!($form instanceof JForm))
 		{
-			return new JException(JText::sprintf('JLIB_FORM_INVALID_FORM_OBJECT', get_class($this)));
+			return new JException(sprintf('Invalid Form Object: :%s', get_class($this)));
```

I think this should "throw" an UnexpectedValueException as well, but you can also put a type hint in the method arguments, e.g.:

```
public function test($element, $value, $group = null, $input = null, JForm $form = null)
```

This is unfortunate design in hindsight because $form should really be mandatory, but we can't do that because it's at the end of the list.